app/style/common/mixins.less

```diff
@@ -17,6 +17,38 @@
  *
  */
 
+// ----------------------------------------------------------------------------
+// THEMES
+// ----------------------------------------------------------------------------
+
+.applyTheme() {
+  @iterateIndexEach: 8;
+  .setColorFade (@name, @fadeOver, @index: 0) when (@index < 96 + @iterateIndexEach) {
+    --@{name}-fade-@{index}: fade(@fadeOver, @index);
+    .setColorFade(@name, @fadeOver, @index + @iterateIndexEach);
+  }
+
+  .setVariables(@list) {
+    .iter(length(@list));
+    .iter(@i) when (@i > 0) {
+      .iter(@i - 1);
+      @pair: extract(@list, @i);
+      @key: extract(@pair, 1);
+      --@{key}: ~'@{@{key}}';
+    }
+  }
+
+  .setVariables(primary secondary);
```

Comment:
I am still not too fond of the naming here. Those aren't `primary` and `secondary` colors, but more like `background` and `foreground`. `primary` and `secondary` can be confusing later on.